GillianCore/engine/Abstraction/VerificationResults.ml

```diff
@@ -0,0 +1,45 @@
+module Json = Yojson.Basic
+module Json_utils = Yojson.Basic.Util
+
+type t = (string * int, bool) Hashtbl.t
+
+let make () : t = Hashtbl.create Config.small_tbl_size
+
+let set_result results proc_name spec_id verified =
+  Hashtbl.replace results (proc_name, spec_id) verified
+
+let merge_results results other_results =
+  let () =
+    Hashtbl.iter
+      (fun id verified -> Hashtbl.replace results id verified)
+      other_results
+  in
+  results
+
+let to_json results =
```

Comment:
What about using `ppx_conv_yojson` instead ?
Maybe in staged mode mode to make it clearer